chore: rework DataSelector styles to css modules

uladzimirdev opened the pull request:
[epic](https://github.com/metabase/metabase/issues/50983)

### Description

PR 3/?

Rework DataSelector to css modules

### How to verify
verify pickers. e.g. follow steps from this [PR](https://github.com/metabase/metabase/pull/50535)

<img width="1005" alt="Screenshot 2024-12-05 at 17 18 18" src="https://github.com/user-attachments/assets/c88cfe2d-6087-49a7-8db9-0cd9166ac88c">

uladzimirdev committed:
chore: rework DataSelector to css modules

uladzimirdev committed:
fix imports

uladzimirdev committed:
Merge branch 'master' into qb-components-css-4

uladzimirdev committed:
revert changes

uladzimirdev committed:
fix styles

uladzimirdev committed:
fix styles

uladzimirdev committed:
fix styles

uladzimirdev committed:
fix styles

uladzimirdev committed:
use upper case

kamilmielnik commented on frontend/src/metabase/query_builder/components/DataSelector/DataSelector.module.css:
```diff
@@ -0,0 +1,45 @@
+.TableSearchContainer {
```
I'm pretty sure our convention is to use `.camelCase`

uladzimirdev replied:
right, but I wanted to have one to one migration from StyledComponents to css modules so it's easier to review when the class name is the same as a component name was

kamilmielnik replied:
That's cool but I think it was not necessary :)

kamilmielnik commented on frontend/src/metabase/query_builder/components/DataSelector/DataSelectorSchemaPicker/DataSelectorSchemaPicker.tsx:
```diff
@@ -36,7 +34,7 @@ const DataSelectorSchemaPicker = ({
 
   return (
-    <Container>
+    <Box w={300}>
```
This `300` comes up a lot in this PR. Maybe we should have a constant for it, WDYT?

uladzimirdev replied:
added a constant and reused it in DataSelector folder

kamilmielnik commented on frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx:
```diff
@@ -123,6 +121,18 @@ export function FieldDataSelector(props) {
   );
 }
 
+export function RawDataBackButton() {
```
nit: It's not used here, can we move it to a a separate file?

uladzimirdev replied:
done

kamilmielnik reviewed (approved):
Works like it used to :+1:

uladzimirdev committed:
address review

uladzimirdev committed:
add missing files

uladzimirdev committed:
one more entry

uladzimirdev commented:
Fixed review comments except class names, I believe some automation can help later with renaming all class names and their usages at once.

Also added 2 more css modules about saved entity picker after review from Kamil, but they are straightforward and shouldn't break anything
